sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -36,6 +36,18 @@ export interface AnalyzeResult {
 // @public
 export type AnalyzeTextOptions = OperationOptions & AnalyzeRequest;
 
+// @public
+export interface AnswerResult {
+    [property: string]: any;
```

Comment:
is the result intended to be unbounded? Or can we use a generic, similar to search results?

Reply:
Actually the swagger says:
```json
"AnswerResult": {
  "properties": {
    "score": {
      "type": "number",
      "format": "double",
      "readOnly": true,
      "description": "The score value represents how relevant the answer is to the the query relative to other answers returned for the query."
    },
    "key": {
      "type": "string",
      "readOnly": true,
      "description": "The key of the document the answer was extracted from."
    },
    "text": {
      "type": "string",
      "readOnly": true,
      "description": "The text passage extracted from the document contents as the answer."
    },
    "highlights": {
      "type": "string",
      "readOnly": true,
      "x-nullable": true,
      "description": "Same text passage as in the Text property with highlighted text phrases most relevant to the query."
    }
  },
  "additionalProperties": true,
  "description": "An answer is a text passage extracted from the contents of the most relevant documents that matched the query. Answers are extracted from the top search results. Answer candidates are scored and the top answers are selected."
}
```

As you can see the `additionalProperties` is set to true. So, the result is intended to be unbounded. We used generics for Search Results. Because we do know that it is going to be of type document. Per the API spec, here it is going to be passages from the document. So, how could we have generics for this one?

Reply:
I wasn't sure if the property name came from the user's schema or if it was entirely arbitrary. From the above swagger, I don't understand why `score`, `key`, and `text` are optional in the response shape? It seems like they will always be returned by the server.

Reply:
Actually, in order for them to be marked as non-optional, the swagger should be:

```json
   .......
   required: ['score', 'key', 'text'],
   .........
   ..........
```

Else, they will be considered optional.

Reply:
ah, yes, you are correct

Reply:
The REST docs don't make these seem optional though: https://docs.microsoft.com/en-us/rest/api/searchservice/preview-api/search-documents#response

Nor does it show any additional properties?

Reply:
I have added the custom transform to mark them as required. (I have created the issue https://github.com/Azure/azure-rest-api-specs/issues/14066 to modify in the swagger permanently. But, it will not happen for this release.) With the latest commit, they are marked as required.